Source/Turbo/Navigator/Routing/Handlers/AppNavigationRouteDecisionHandler.swift

```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+final class AppNavigationRouteDecisionHandler: RouteDecisionHandler {
+    let name: String = "app-navigation"
+
+    func matches(location: URL,
+                 configuration: Navigator.Configuration) -> Bool {
+        if #available(iOS 16, *) {
+            return configuration.startLocation.host() == location.host()
+        }
+
+        return configuration.startLocation.host == location.host
+    }
+
+    func handle(location: URL,
+                configuration: Navigator.Configuration,
+                navigator: Navigator) -> Router.Decision {
+        return .navigate
+    }
+}
```

Comment:
If we want developers to use the handlers then we need to make them public, along with an explicit `init()`. Same goes for the other two handlers.

```suggestion
public final class AppNavigationRouteDecisionHandler: RouteDecisionHandler {
    public let name: String = "app-navigation"

    public init() {}

    public func matches(location: URL,
                 configuration: Navigator.Configuration) -> Bool {
        if #available(iOS 16, *) {
            return configuration.startLocation.host() == location.host()
        }

        return configuration.startLocation.host == location.host
    }

    public func handle(location: URL,
                configuration: Navigator.Configuration,
                navigator: Navigator) -> Router.Decision {
        return .navigate
    }
}
```

Reply:
Good point!